addons/project/models/project_task.py

```diff
@@ -817,21 +819,33 @@ def copy_data(self, default=None):
 
             if not default.get('stage_id'):
                 vals['stage_id'] = task.stage_id.id
-            if 'active' not in default and not task['active'] and not self.env.context.get('copy_project'):
+            if 'active' not in default and not task['active'] and not self.env.context.get('copy_project') and not self.env.context.get('has_template_ancestor'):
                 vals['active'] = True
-            vals['name'] = task.name if self.env.context.get('copy_project') else _("%s (copy)", task.name)
+            vals['name'] = task.name if self.env.context.get('copy_project') or self.env.context.get('copy_from_template') else _("%s (copy)", task.name)
             if task.recurrence_id and not default.get('recurrence_id'):
                 vals['recurrence_id'] = task.recurrence_id.copy().id
             if task.allow_milestones:
                 vals['milestone_id'] = milestone_mapping.get(vals['milestone_id'], vals['milestone_id'])
-            if task.child_ids and not default.get('child_ids'):
+            copy_from_template = task.is_template and self.env.context.get('copy_from_template')
+            active_test = self.env.context.get('active_test', True) and not copy_from_template
+            if not default.get('child_ids') and task.with_context(active_test=active_test).child_ids:
+                context = {
+                    'active_test': active_test,
+                }
                 default = {
                     'parent_id': False,
                 }
-                vals['child_ids'] = [Command.create(child_id.copy_data(default)[0]) for child_id in task.child_ids]
+                if copy_from_template:
+                    default['active'] = True
+                if task.is_template:
+                    context['has_template_ancestor'] = True
+                vals['child_ids'] = [Command.create(child_id.copy_data(default)[0]) for child_id in task.with_context(**context).child_ids]
             if not has_default_users and vals['user_ids']:
                 active_users = task.user_ids & active_users
                 vals['user_ids'] = [Command.set(active_users.ids)]
+            if task.is_template and not self.env.context.get('copy_from_template'):
```

Comment:
```suggestion
            if copy_from_template:
```
? 🤔

Reply:
Not the same, `copy_from_template` is `task.is_template and self.env.context.get('copy_from_template')`